In [1]:
import pandas as pd
import numpy as np
from selenium.common.exceptions import NoSuchElementException
import time
# Importing selenium webdriver 
from selenium import webdriver

#Importing requests
import requests

# importing regex
import re

In [2]:
import warnings
warnings.filterwarnings('ignore')

In [13]:
# Activating the chrome browser
driver=webdriver.Chrome("E:\Data science\FLIPROBO\web driver\chromedriver.exe") 
time.sleep(3)

In [14]:
#opening the cars24 website
url = 'https://bit.ly/3HbOtY7'
driver.get(url)

In [19]:
driver.find_element_by_xpath("//button[@class='btn _3BmFU AgO6q']").click()

In [20]:
#Creating empty lists
brand=[]
year=[]
fuel=[]
transmission=[]
km=[]
price=[]
prod_URL=[]

In [22]:
#Getting the URL of the cars
for i in range(0,500):
    url=driver.find_elements_by_xpath("//a[@class='_9Ue0B']")
    for i in url:
        prod_URL.append(i.get_attribute('href'))
    try:
        next_btn = driver.find_element_by_xpath("//button[@class='btn _3BmFU AgO6q']").click()
    except NoSuchElementException:
        pass

In [23]:
prod_URL

['https://www.cars24.com/buy-used-maruti-swift-2012-cars-new-delhi-1124310622/',
 'https://www.cars24.com/buy-used-maruti-swift-2012-cars-new-delhi-1000969248/',
 'https://www.cars24.com/buy-used-maruti-swift-dzire-2013-cars-faridabad-1125443125/',
 'https://www.cars24.com/buy-used-toyota-innova-2013-cars-new-delhi-1012004420/',
 'https://www.cars24.com/buy-used-maruti-swift-dzire-2013-cars-faridabad-1008854220/',
 'https://www.cars24.com/buy-used-maruti-alto-k10-2014-cars-faridabad-1015689266/',
 'https://www.cars24.com/buy-used-maruti-swift-dzire-2014-cars-new-delhi-1018057844/',
 'https://www.cars24.com/buy-used-maruti-swift-2015-cars-new-delhi-1009658243/',
 'https://www.cars24.com/buy-used-maruti-alto-k10-2019-cars-new-delhi-1006618561/',
 'https://www.cars24.com/buy-used-maruti-swift-2013-cars-new-delhi-1001623644/',
 'https://www.cars24.com/buy-used-hyundai-i20-2013-cars-faridabad-1005662020/',
 'https://www.cars24.com/buy-used-maruti-swift-2015-cars-noida-1006609108/',
 'https:

In [24]:
len(prod_URL)

55193

In [25]:
for i in prod_URL:
    driver.get(i) 
    
#Extracting car name
    try:
        brand.append(driver.find_element_by_xpath("//h2[@class='_3FpCg']").text)
    except NoSuchElementException as e:
        brand.append("-")
        
#Extracting the transmission
    try:
        transmission.append(driver.find_element_by_xpath("//p[@class='cvakB']/span").text)  
    except NoSuchElementException as e:
        transmission.append("-")
        
#Extracting the fuel consumed
    try:
        fuel.append(driver.find_element_by_xpath("//ul[@class='bVR0c']/li[3]").text)
    except NoSuchElementException as e:
        fuel.append("-")
        
#Extracting km driven
    try:
        km.append(driver.find_element_by_xpath("//ul[@class='bVR0c']/li[1]").text)
    except NoSuchElementException as e:
        km.append("-") 
        

        #Extracting the price details
    try:
        price.append(driver.find_element_by_xpath("//div[@class='_7udZZ']").text.replace('₹',''))
    except NoSuchElementException as e:
        price.append("-")

ERROR:root:Internal Python error in the inspect module.
Below is the traceback from this internal error.



Traceback (most recent call last):
  File "C:\Users\HP\anaconda3\lib\site-packages\IPython\core\interactiveshell.py", line 3437, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-25-5efff5924350>", line 2, in <module>
    driver.get(i)
  File "C:\Users\HP\anaconda3\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 436, in get
    self.execute(Command.GET, {'url': url})
  File "C:\Users\HP\anaconda3\lib\site-packages\selenium\webdriver\remote\webdriver.py", line 422, in execute
    response = self.command_executor.execute(driver_command, params)
  File "C:\Users\HP\anaconda3\lib\site-packages\selenium\webdriver\remote\remote_connection.py", line 421, in execute
    return self._request(command_info[0], url, body=data)
  File "C:\Users\HP\anaconda3\lib\site-packages\selenium\webdriver\remote\remote_connection.py", line 443, in _request
    resp = self._conn.request(method, url, body=body, headers=headers)
  File "C:\Users\HP\anaconda3\

TypeError: object of type 'NoneType' has no len()

In [32]:
year = []
brand1 = []
for i in brand:
    year.append(i.split(' ')[0])
year

['2013',
 '2016',
 '2014',
 '2018',
 '2014',
 '2013',
 '2019',
 '2019',
 '2014',
 '2014',
 '2014',
 '2019',
 '2013',
 '2014',
 '2020',
 '-',
 '2014',
 '2017',
 '2015',
 '2013',
 '2015',
 '2019',
 '2019',
 '2019',
 '2013',
 '2019',
 '2017',
 '2017',
 '2014',
 '2019',
 '2015',
 '-',
 '2017',
 '2013',
 '2013',
 '2017',
 '2019',
 '2014',
 '2013',
 '2013',
 '2016',
 '2019',
 '2015',
 '2020',
 '2020',
 '2015',
 '2019',
 '2013',
 '-',
 '2014',
 '2017',
 '2015',
 '2014',
 '2017',
 '2016',
 '2019',
 '2018',
 '2018',
 '2017',
 '2014',
 '2013',
 '2015',
 '2014',
 '2018',
 '2014',
 '2013',
 '2019',
 '2019',
 '2014',
 '2015',
 '2014',
 '2019',
 '2013',
 '2013',
 '2020',
 '2019',
 '2014',
 '2017',
 '2015',
 '2013',
 '2015',
 '2017',
 '2019',
 '2019',
 '2014',
 '2019',
 '2019',
 '2019',
 '2013',
 '2019',
 '2015',
 '2013',
 '2017',
 '2013',
 '2013',
 '2017',
 '2019',
 '2014',
 '2013',
 '2015',
 '2016',
 '2019',
 '2014',
 '2020',
 '2020',
 '2015',
 '2019',
 '2013',
 '2017',
 '2017',
 '2017',
 '2015',
 

In [33]:
car=pd.DataFrame()
car['Brand'] = brand
car['Transmission'] = transmission
car['Fuel'] = fuel
car['Kms'] = km
car['Price'] = price
car['year'] = year
car

Brand Transmission    Fuel        Kms      Price  \
0             2013 Maruti Swift       Manual  Diesel  65,561 km   3,32,699   
1             2016 Maruti Swift       Manual  Petrol  58,818 km   4,07,599   
2       2014 Maruti Swift Dzire       Manual  Diesel  55,542 km   3,43,799   
3     2018 Maruti Vitara Brezza       Manual  Diesel  71,192 km   6,74,099   
4       2014 Maruti Swift Dzire       Manual  Diesel  55,542 km   3,43,799   
...                         ...          ...     ...        ...        ...   
4358   2013 Hyundai New Elantra       Manual  Diesel  47,959 km   6,60,999   
4359          2013 BMW 5 Series    Automatic  Diesel  46,477 km  18,17,799   
4360      2016 Mahindra Scorpio       Manual  Diesel  51,588 km   8,75,899   
4361         2019 Hyundai VENUE       Manual  Petrol  31,118 km   8,67,899   
4362          2015 Maruti Swift       Manual  Diesel  46,168 km   4,16,499   

      year  
0     2013  
1     2016  
2     2014  
3     2018  
4     2014  
...    ...  
4358  2013  
4359  2013  
4360  2016  
4361  2019  
4362  2015  

[4363 rows x 6 columns]

### we have got a 4363 rows and 5 columns for our dataset

In [34]:
#Saving in a csv file
car.to_csv('cars24data.csv')

In [35]:
pd.read_csv('cars24data.csv')

Unnamed: 0                      Brand Transmission    Fuel        Kms  \
0              0          2013 Maruti Swift       Manual  Diesel  65,561 km   
1              1          2016 Maruti Swift       Manual  Petrol  58,818 km   
2              2    2014 Maruti Swift Dzire       Manual  Diesel  55,542 km   
3              3  2018 Maruti Vitara Brezza       Manual  Diesel  71,192 km   
4              4    2014 Maruti Swift Dzire       Manual  Diesel  55,542 km   
...          ...                        ...          ...     ...        ...   
4358        4358   2013 Hyundai New Elantra       Manual  Diesel  47,959 km   
4359        4359          2013 BMW 5 Series    Automatic  Diesel  46,477 km   
4360        4360      2016 Mahindra Scorpio       Manual  Diesel  51,588 km   
4361        4361         2019 Hyundai VENUE       Manual  Petrol  31,118 km   
4362        4362          2015 Maruti Swift       Manual  Diesel  46,168 km   

          Price  year  
0      3,32,699  2013  
1      4,07,599  2016  
2      3,43,799  2014  
3      6,74,099  2018  
4      3,43,799  2014  
...         ...   ...  
4358   6,60,999  2013  
4359  18,17,799  2013  
4360   8,75,899  2016  
4361   8,67,899  2019  
4362   4,16,499  2015  

[4363 rows x 7 columns]